homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +170,182 @@ def update_state(self, new_state):
+
+    def set_on_off(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "on_off" to %s',
+                      self.entity_id, value)
+        self._flag_active = True
+        service = SERVICE_TURN_ON if value else SERVICE_TURN_OFF
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_play_pause(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "play_pause" to %s',
+                      self.entity_id, value)
+        service = SERVICE_MEDIA_PLAY if value else SERVICE_MEDIA_PAUSE
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_toggle_mute(self, value):
+        """Move switch state to value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set switch state for "toggle_mute" to %s',
+                      self.entity_id, value)
+        self._flag_toggle_mute = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_MEDIA_VOLUME_MUTED: value}
+        self.call_service(DOMAIN, SERVICE_VOLUME_MUTE, params)
+
+    def set_volume_step(self, value):
+        """Send volume step value if call came from HomeKit."""
+        _LOGGER.debug('%s: Step volume by %s',
+                      self.entity_id, value)
+
+        service = SERVICE_VOLUME_DOWN if value else SERVICE_VOLUME_UP
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_input_source(self, value):
+        """Send input set value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set active_identifer to %s',
+                      self.entity_id, value)
+
+        source = self.sources[value]
+        self._flag_active_identifier = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_INPUT_SOURCE: source}
+        self.call_service(DOMAIN, SERVICE_SELECT_SOURCE, params)
+
+    def set_remote_key(self, value):
+        """Send remote key value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set remote key to %s', self.entity_id, value)
+
+        if value in MEDIA_PLAYER_KEYS:
+            service = MEDIA_PLAYER_KEYS[value]
+            if service == SERVICE_MEDIA_PLAY_PAUSE:
+                state = self.hass.states.get(self.entity_id).state
+                if state in (STATE_PLAYING, STATE_PAUSED):
+                    self.set_play_pause(state == STATE_PAUSED)
+                    return
+            params = {ATTR_ENTITY_ID: self.entity_id}
+            self.call_service(DOMAIN, service, params)
+
+    def update_state(self, new_state):
+        """Update Television state after state changed."""
+        current_state = new_state.state
+
+        if self.char_active:
+            hk_state = current_state not in (STATE_OFF, STATE_UNKNOWN, 'None')
```

Comment:
Does `'None'` ever happen? Isn't that covered with `STATE_UNKNOWN`?

Reply:
I have removed the `'None'` state. This should be covered by `STATE_UNKNOWN`

Reply:
I did see this was also used in the normal `MediaPlayer` accessory.